tfx/components/base/base_executor_test.py

```diff
@@ -18,40 +18,153 @@
+                    exec_properties: Dict[Text, Union[int, float, Text]]
+                    ) -> Dict[Text, beam.pvalue.PCollection]:
+    beam_outputs = {_SHUFFLED_EXAMPLES: (beam_inputs[_EXAMPLES]
+                                         | beam.Map(str.strip)
+                                         | beam.Map(int)
+                                         | beam.CombineGlobally(sum))}
+    return beam_outputs
+
+  def write_outputs(self, pipeline: beam.Pipeline,
+                    beam_outputs: Dict[Text, beam.pvalue.PCollection],
+                    input_dict: Dict[Text, List[types.Artifact]],
+                    output_dict: Dict[Text, List[types.Artifact]],
+                    exec_properties: Dict[Text, Union[int, float, Text]]
+                    ) -> None:
+    output_dir = output_dict[_SHUFFLED_EXAMPLES].uri
+    _ = (beam_outputs[_SHUFFLED_EXAMPLES]
+         | beam.io.textio.WriteToText(output_dir))
+
+  def Do(self, input_dict: Dict[Text, List[types.Artifact]],
+         output_dict: Dict[Text, List[types.Artifact]],
+         exec_properties: Dict[Text, Union[int, float, Text]]) -> None:
+    with self._make_beam_pipeline() as p:
+      beam_inputs = self.read_inputs(p, input_dict, output_dict,
+                                     exec_properties)
+      beam_outputs = self.run_component(p, beam_inputs, input_dict,
+                                        output_dict, exec_properties)
+      self.write_outputs(p, beam_outputs, input_dict, output_dict,
+                         exec_properties)
+
 class BaseExecutorTest(tf.test.TestCase):
 
   def testBeamSettings(self):
     executor_context = base_executor.BaseExecutor.Context(
         beam_pipeline_args=['--runner=DirectRunner'])
     executor = _TestExecutor(executor_context)
-    options = executor._make_beam_pipeline().options.view_as(StandardOptions)
+    options = executor._make_beam_pipeline().options.view_as(StandardOptions) # pylint: disable=protected-access
     self.assertEqual('DirectRunner', options.runner)
 
     executor_context = base_executor.BaseExecutor.Context(
         beam_pipeline_args=['--direct_num_workers=2'])
     executor = _TestExecutor(executor_context)
-    options = executor._make_beam_pipeline().options.view_as(DirectOptions)
+    options = executor._make_beam_pipeline().options.view_as(DirectOptions) # pylint: disable=protected-access
     self.assertEqual(2, options.direct_num_workers)
 
 
+class BeamExecutorTest(tf.test.TestCase):
+
+  def setUp(self):
+    super(BeamExecutorTest, self).setUp()
+    self._input_data_path = (
+        os.path.join(os.environ.get(
+            'TEST_UNDECLARED_OUTPUTS_DIR', self.get_temp_dir()), 'input.txt'))
+    output_data_path = (
+        os.path.join(os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR',
+                                    self.get_temp_dir()), 'output.txt'))
+    self._full_output_data_path = output_data_path + '-00000-of-00001'
+
+    self._generate_data(self._input_data_path)
```

Comment:
nit: delete this file in tearDown()

Reply:
I tried doing that, but I don't think there is a need to since the `super(BeamExecutorTest, self).tearDown()`method is automatically called and it removes `self._full_output_data_path` since it is a tmp file anyways. I got an error that said:

`Traceback (most recent call last):
  File "tfx/components/base/base_executor_test.py", line 144, in tearDown
    os.remove(self._full_output_data_path)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/base_executor_testij6772g8/tmprgxyinl3/output.txt-00000-of-00001'`